tests/tests_geomstats/test_hyperbolic.py

```diff
@@ -18,478 +19,402 @@
+                    y_intrinsic=gs.array([1, 6.0]),
+                ),
+                dict(
+                    dim=4,
+                    x_intrinsic=gs.array([10, 0.2, 3, 4]),
+                    y_intrinsic=gs.array([1, 6, 2.0, 1]),
+                ),
+            ]
+            return self.generate_tests(smoke_data)
 
-        result = self.metric.inner_product(tangent_vec_a, tangent_vec_b, base_point)
+    testing_data = TestDataHyperboloidMetric()
 
-        expected = minkowski_space.metric.inner_product(
+    def test_inner_product_is_minkowski_inner_product(
+        self, dim, tangent_vec_a, tangent_vec_b, base_point
+    ):
+        metric = self.metric(dim)
+        minkowki_space = Minkowski(dim + 1)
+        result = metric.inner_product(tangent_vec_a, tangent_vec_b, base_point)
+        expected = minkowki_space.metric.inner_product(
             tangent_vec_a, tangent_vec_b, base_point
         )
-
-        self.assertAllClose(result, expected)
-
-    def test_squared_norm_and_squared_dist(self):
-        """
-        Test that the squared distance between two points is
-        the squared norm of their logarithm.
-        """
-        point_a = gs.array([2.0, 1.0, 1.0, 1.0])
-        point_b = gs.array([4.0, 1.0, 3.0, math.sqrt(5)])
-        log = self.metric.log(point=point_a, base_point=point_b)
-        result = self.metric.squared_norm(vector=log)
-        expected = self.metric.squared_dist(point_a, point_b)
-
-        self.assertAllClose(result, expected)
-
-    def test_norm_and_dist(self):
-        """
-        Test that the distance between two points is
-        the norm of their logarithm.
-        """
-        point_a = gs.array([2.0, 1.0, 1.0, 1.0])
-        point_b = gs.array([4.0, 1.0, 3.0, math.sqrt(5)])
-        log = self.metric.log(point=point_a, base_point=point_b)
-        result = self.metric.norm(vector=log)
-        expected = self.metric.dist(point_a, point_b)
-
-        self.assertAllClose(result, expected)
-
-    def test_log_and_exp_edge_case(self):
-        """
-        Test that the Riemannian exponential
-        and the Riemannian logarithm are inverse.
-
-        Expect their composition to give the identity function.
-        """
-        # Riemannian Log then Riemannian Exp
-        # Edge case: two very close points, base_point_2 and point_2,
-        # form an angle < epsilon
-        base_point_intrinsic = gs.array([1.0, 2.0, 3.0])
-        base_point = self.space.from_coordinates(base_point_intrinsic, "intrinsic")
-        point_intrinsic = base_point_intrinsic + 1e-12 * gs.array([-1.0, -2.0, 1.0])
-        point = self.space.from_coordinates(point_intrinsic, "intrinsic")
-
-        log = self.metric.log(point=point, base_point=base_point)
-        result = self.metric.exp(tangent_vec=log, base_point=base_point)
-        expected = point
-
-        self.assertAllClose(result, expected)
-
-    def test_exp_and_log_and_projection_to_tangent_space_general_case(self):
```

Comment:
missing?

Reply:
nope already of part of exp-log composition